addon/model.js

```diff
@@ -145,6 +146,16 @@ if (CUSTOM_MODEL_CLASS) {
   }
 }
 
+// If a value returned from unknownProperty is a ManagedArray we need to access '[]'
+// to keep parity with how ember treats arrays
+// see https://github.com/emberjs/ember.js/blob/3ce13cea235cde8a87d89473533c453523412764/packages/%40ember/-internals/metal/lib/property_get.ts#L136
+function accessBracketsIfArray(maybeArray) {
+  if (MANAGED_ARRAYS.has(maybeArray)) {
+    get(maybeArray, '[]');
+  }
+  return maybeArray;
+}
```

Comment:
Let's rename this to something like `readCachedValue(key)`

As it stands the naming is odd. You get usage like

```js
  if (key in this._cache) {
      return accessBracketsIfArray(this._cache[key]);
    }
```

So here the call site cares about the return value, but is calling a function named for its side-effects. This is a kind of abstraction leakage -- contrast to

```js
  if (key in this._cache) {
      return readCachedValue(this._cache, key);
    }
```

now the call site is asking for what it needs in order to return a value (the cached value) and the side effects necessary to entangle the array via ember's semantics is encapsulated in the implementation of `readCachedValue`.

Reply:
done

Reply:
what's the reason for preferring `readCachedValue(value)` over the API I suggested above?

Reply:
Discussed on slack, updated the method now so we don't have a separate function call